experimental/dds/tree2/src/test/utils.ts

```diff
@@ -408,6 +409,46 @@ export function spyOnMethod(
 	};
 }
 
+/**
+ * @returns `true` iff the given delta has a visible impact on the document tree.
+ */
+export function isDeltaVisible(delta: Delta.MarkList): boolean {
```

Comment:
Is it difficult to avoid creating "invisible" deltas?

Reply:
In general, yes, because it might require following a sequence of moves to see if they span the boundary between transient and non-transient stuff.

For the tests that use this function, we could avoid it in more cases if `NodeChangeComposer` had the ability to return undefined in order to signify that the outcome of the composition turned out to be a no-op.